sys/freebsd/socket.txt

```diff
@@ -58,7 +57,7 @@ send_msghdr {
 	msg_iovlen	len[msg_iov, intptr]
 	msg_control	ptr[in, array[cmsghdr]]
 	msg_controllen	bytesize[msg_control, intptr]
-	msg_flags	flags[send_flags, int32]
+	msg_flags	int32
```

Comment:
Why? This field contains some flags and we had some enumeration for them, but now we have just int.

Reply:
Interesting. I guess I always confused send flags and msghdr flags, so we seem to have the same problem on linux.

A better way to describe this then is `const[0, int32]`. It makes it visible that one wasn't just lazy to detail what flags are there and it makes fuzzing more efficient as it will not spend iterations for mutation of flags.

Reply:
Fixed in https://github.com/google/syzkaller/pull/857/commits/0b5ce8a432cdf7a6e7876519e0bc78b4531ee82b.

I think the other OSes need a similar fix. Please note that I also changed the `msg` parameter of the `recvmsg()` call from `in` to `inout`.

BTW: Is there a type for enumeration values, like it would be appropriate for `optname` in `setsockopt()` or `getsockopt()`. Currently `flags` is used, but I guess that is a type where you can combine individual values by `or`.

Reply:
I can provide a patch for the other OSes, of course, once this is merged (to avoid rebasing).

Reply:
> I can provide a patch for the other OSes

If you don't mind, it would be good.

Reply:
> BTW: Is there a type for enumeration values

We use flags for both, but we have bitmask autodetection:

https://github.com/google/syzkaller/blob/master/pkg/compiler/types.go#L254

If we go with separate type, say, it becomes impossible to handle when, say, low part of a field is flags, but high part is plain value. If we do auto detection, it's both easier for user and allows to potentially detect and handle such more complex cases.